### PR TITLE
Frontend cleanup

### PR DESCRIPTION
**Pull request by willmt80**
Added line breaks to components for readability. Moved some of functionality out of loadMap. Changed many uses of the word 'street' to the word 'block. 

Currently blocked by this speak-for-the-trees-backend pr https://github.com/Code-4-Community/speak-for-the-trees-backend/pull/51

### CODE REVIEW

File: src/components/MapComponent.vue [hunk 1]
```diff
@@ -0,0 +1,335 @@
+<template>
+  <div id="map-component">
```
Comment (jackblanc) [inline src/components/MapComponent.vue:2]: 🌱 How much effort would it take to have a satellite view on the map? I feel like that's much more fitting.

File: src/components/MapComponent.vue [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    ...mapState({
+      reservedBlocks: 'reservedBlocks',
+    }),
+
+    // Determines which blocks to display depending on the user is looking at
+    // reserve new blocks, current reservations, or admin map
+    blockFilter() {
+      if (this.isAdminMap) {
+        // Admins can see all current reserved blocks by any user
+        return 'RESERVED = 1 OR RESERVED = 2';
```
Comment (jackblanc) [inline src/components/MapComponent.vue:126]: 🌱 I think admins would like being able to see unreserved blocks too

File: src/components/MapComponent.vue [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+    z-index: 100;
+    right: 5px;
+    top: 5px;
+    float: right;
+    background: #9AC356;
+    color: white;
+    border: none;
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+  }
+</style>
```
Comment (jackblanc) [inline src/components/MapComponent.vue:335]: Is there any way we could reduce the size of this file and abstract things to other files?

File: src/components/SelectedBlocks.vue [hunk 4]
```diff
@@ -5,26 +5,26 @@
         class="trigger"
         :id="title"
         @click="popoverToggle = !popoverToggle"
-        :disabled="streets.length === 0">
+        :disabled="blocks.length === 0">
```
Comment (jackblanc) [inline src/components/SelectedBlocks.vue:8]: 🔥

File: src/components/Board.vue [hunk 5]
```diff
@@ -21,13 +21,16 @@
 
 <script>
 export default {
+
```
[the remark this answers is not in the document]
Comment (willmt80) [reply src/components/Board.vue:24]: It's not lint enforced, I just find it's much easier to read files with line spaces between sections/methods etc.

File: src/components/MapComponent.vue (hunk 3 above)
Comment (willmt80) [reply src/components/MapComponent.vue:335]: Yeah at least all of the constants can be moved. There could maybe be some more complicated component rearrangement but that might be its own ticket.

File: src/components/MapComponent.vue (hunk 2 above)
Comment (willmt80) [reply src/components/MapComponent.vue:126]: Should admins be able to assign an unreserved block to themselves/any other user? Adding ether of these features could be a good ticket for the new members who are going to work with the map.

File: src/components/MapComponent.vue [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    blockActions() {
+      if (this.reservedFilter === 0) {
+        return [mapConstants.ADD_TO_RESERVE];
+      }
+      return [mapConstants.ADD_TO_RELEASE, mapConstants.ADD_TO_COMPLETE];
+    },
+
+    // contains the information about each of the blocks, each block is a feature.
+    // https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-FeatureLayer.html
+    blockLayerObject() {
```
Comment (willmt80) [inline src/components/MapComponent.vue:87]: Before, complete blocks and reserved blocks had separate renderers on the admin map. Now all blocks are displayed from this featureLayer

File: src/components/MapComponent.vue [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+        // If there are given IDs, have the map only show the given blocks
+        return `ID = ${this.reservedBlocks.join(' OR ID = ')} AND RESERVED = 1`;
+      }
+      // Will not display any blocks
+      return '1=0';
+    },
+
+    // The content of the popup when clicking on a block
+    // https://developers.arcgis.com/javascript/latest/api-reference/esri-PopupTemplate.html
+    blockPopupTemplate() {
```
Comment (willmt80) [inline src/components/MapComponent.vue:66]: popuptemplate and blockactions were in the originally consts in the loadmap() method, moved to separate computed properties

File: src/components/MapComponent.vue [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+    <b-button class="label" type="submit" @click="showLabels">
+      {{`${(this.labelsVisible ? "Hide" : "Show")} Block Numbers`}}
+    </b-button>
+  </div>
+</template>
+
+<script>
+import { mapState } from 'vuex';
+import { loadModules } from 'esri-loader';
+import mapConstants from '../constants/mapConstants';
```
Comment (willmt80) [inline src/components/MapComponent.vue:12]: Moved unchanging objects that do not rely on ArcGIS lazy load to mapConstants to remove some clutter.

File: src/components/MapComponent.vue [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+                  }
+                  this.view.popup.open({
+                    features: blockFeatures,
+                  });
+                });
+            }
+          });
+
+          // Hides the completed block action when a block is completed
+          this.view.popup.watch('selectedFeature', (graphic) => {
```
Comment (willmt80) [inline src/components/MapComponent.vue:193]: This replaces the need for the separate layer for the completed blocks. Disables the complete action if the selected block has a reservation status of 2

**Review by willmt80 — COMMENTED** [review]
Names of file changed so here are a few comments with the most significant changes

File: src/constants/mapConstants.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
+// The action button to add a block to the reserve list
```
Comment (willmt80) [inline src/constants/mapConstants.js:1]: All of this used to be in mapComponent.vue

**Comment by willmt80** [comment]
Changed base map from the boring grey to a satellite view because it feels more tree-y.

satellite map

File: src/components/MapComponent.vue (hunk 2 above)
Comment (jackblanc) [reply src/components/MapComponent.vue:52]: Yeah that would be great, can you write a ticket stub? cc @florisdobber

**Review by jackblanc — APPROVED** [review]
This is awesome thank you!

File: src/components/MapComponent.vue (hunk 2 above)
Comment (florisdobber) [reply src/components/MapComponent.vue:52]: Yup done

**Pull request merged**